src/lib/operations.ts

```diff
@@ -117,6 +121,122 @@ const prepareToEditExtrude: PrepareToEditCallback =
     }
   }
 
+/**
+ * Gather up the argument values for the Extrude command
+ * to be used in the command bar edit flow.
+ */
+const prepareToEditShell: PrepareToEditCallback =
+  async function prepareToEditShell({ operation }) {
+    const baseCommand = {
+      name: 'Shell',
+      groupId: 'modeling',
+    }
+
+    if (
+      operation.type !== 'StdLibCall' ||
+      !operation.labeledArgs ||
+      !operation.unlabeledArg ||
+      operation.unlabeledArg.value.type !== 'Solid' ||
+      !('thickness' in operation.labeledArgs) ||
+      !('faces' in operation.labeledArgs) ||
+      !operation.labeledArgs.thickness ||
+      !operation.labeledArgs.faces ||
+      operation.labeledArgs.faces.value.type !== 'Array'
+    ) {
+      return baseCommand
+    }
+
+    // Build an artifact map here of eligible artifacts corresponding to our current sweep
+    // that we can query in another loop later
```

Comment:
The logic below is fairly verbose and not very straightforward, super open to suggestion as well.

Reply:
I don't have any suggestions yet, I think you've done this well with what we have. Having all these `prepare*` functions together should help us see the pattern between them and suggest what we need. I think either an additional API that takes that artifactGraph and the operation, or something provided from the operation itself, but I'm not sure yet.